src/Processors/Chunk.h

```diff
@@ -11,6 +11,9 @@ class ChunkInfo
 public:
     virtual ~ChunkInfo() = default;
     ChunkInfo() = default;
+
+    /// To be overriden by MergeTreePartLevelInfo
+    virtual UInt64 getMergeTreePartLevel() const { return 0; }
```

Comment:
Avoid changing of Chunk and ChunkInfo interfaces.
Use typeid_cast or assert_cast to descendant if needed.
Examples:
https://github.com/ClickHouse/ClickHouse/blob/e493789bf346d4521391de925f39b9da0d929866/src/Processors/Transforms/AggregatingTransform.h#L22-L28

https://github.com/ClickHouse/ClickHouse/blob/e493789bf346d4521391de925f39b9da0d929866/src/Processors/Transforms/AggregatingTransform.cpp#L43-L54

Reply:
I've seen previous discussion, but I suppose typeid_cast should be the same cost as calling the virtual call.
Probably even cheaper, cause the check itself is vtable pointers comparison, and function can be inlined after.

Btw we can do this check once per chunk and save the result somewhere.

Reply:
Technically speaking, we only need to calculate it once per input stream of `IMergingAlgorithm`.